### DISCUSSION TITLE
How to mute text betwen asterisks ?

### DISCUSSION THREAD

**Discussion by Wuzzooy** [KoljaB/RealtimeTTS#157] [Q&A] [answered]
Hello,
Thank you for your work, i really appreciate it.
I have no experience in coding and i would like to know how to proceed and which file to modify to not send text between *asterisks* please.
It's for when LLM write his action in roleplay mode.
I'm using coqui engine.

Edit: can be done in steam2sentence.py file using regex apparently, not sure if it's the right way though.
```
text = re.sub(r'\*[^*]*\*', '', text)
```

**Comment by KoljaB** [comment, marked as answer]
Two ways I can think of to achieve this:

1. Change CharIterator in threadsafe_generators.py
    - check each character in the __next__ method for *
    - remember if a * occured and discard all chars (not return them) until the next * occurs 
2. Write an own generator to wrap the generator from the LLM 
    - retrieve the token from the LLM
    - check the token char by char for *, again remember and not yield in between *s
    - yield all chars from that token outside of *s

**Reply by KoljaB** [reply]
Yes, stream2sentence.py is also an option, you'd have to check every sentence for * then. I don't think a regex is a good way to filter them out, because a * could occur in one sentence and the terminating * in the next one.

**Comment by Wuzzooy** [comment]
Thank you very much for your insights.
